chore: Add pre-release nightly demo testing issue template (#609)
stackabletech/issues: .github/ISSUE_TEMPLATE/pre-release-demos-nightly.md

---
name: Pre-Release Demo Upgrade Testing from Stable to Nightly
about: This template can be used to track the upgrade testing of demos from stable to nightly leading up to the next Stackable release
title: "chore(tracking): Test demos on nightly versions"
labels: ['epic']
assignees: ''
---

<!--
    Make sure to update the link in '.github/ISSUE_TEMPLATE/release.md' when
    you change the front matter above.
-->

<!--
    DO NOT REMOVE THIS COMMENT. It is intended for people who might copy/paste from the previous release issue.
    This was created by an issue template: https://github.com/stackabletech/issues/issues/new/choose.
-->

## Pre-Release Demo Testing on Nightly

Part of <https://github.com/stackabletech/issues/issues/TRACKING_ISSUE>

For each demo, run the following commands:

> [!NOTE]
> Record any issues or anomalies during the process in a comment on this issue.
> Eg:
> ```
> :green_circle: **airflow-scheduled-job**
>
> The CRD had been updated and I needed to change the following in the manifest:
> ...
> ```

> ```shell

Reviewer: **nit(non-blocker)**: this should be `sh` (or `bash` if necessary).

~~~suggestion
```sh
~~~

_you don't need to change it now, but I probably will when I next see it_

Author: Decided to leave as is.

---

> name: Pre-Release Demo Testing on Nightly
> about: This template can be used to track the testing of demos on nightly leading up to the next Stackable release

Reviewer: This is also testing the upgrade from stable to nightly, so maybe

```suggestion
name: Pre-Release Demo Upgrade Testing from Stable to Nightly
about: This template can be used to track the uprade testing of demos from stable to nightly leading up to the next Stackable release
```

title: "chore(tracking): Test demos on nightly versions"
labels: ['epic']
assignees: ''
---

<!--
    Make sure to update the link in '.github/ISSUE_TEMPLATE/release.md' when
    you change the front matter above.
-->

<!--
    DO NOT REMOVE THIS COMMENT. It is intended for people who might copy/paste from the previous release issue.
    This was created by an issue template: https://github.com/stackabletech/issues/issues/new/choose.
-->

## Pre-Release Demo Testing on Nightly

Part of <https://github.com/stackabletech/issues/issues/TRACKING_ISSUE>

> 
> For each demo, run the following commands:
> 

Reviewer: ```suggestion

For each demo, run the following commands:

> [!NOTE]
> Record any issues or anomalies during the process in a comment on this issue.
> Eg:
> ```
> :green_circle: **airflow-scheduled-job**
>
> The CRD had been updated and I needed to change the following in the manifest:
> ...
> ```

```

```shell

> # Install demo

Reviewer: ```suggestion
# Install demo (stable)
```

> stackablectl demo <DEMO_NAME>

Reviewer: ```suggestion
stackablectl demo install <DEMO_NAME>
```

> # Uninstall operators
> stackablectl release uninstall <CURRENT_RELEASE>

Reviewer: ```suggestion
# Get a list of operators that are installed
kubectl -n stackable-operators get deploy -o json | jq -r '.items[].metadata.name' | sed -e 's/-operator-deployment//' | sort

# Uninstall operators
stackablectl release uninstall <CURRENT_RELEASE>
```

Author: Or, just use `stackablectl operator installed` :)

Author: Done in 3a9ef57

Reviewer: True, although for me there is too much noise.

Perhaps `stackablectl operator installed --output=plain` is a nice compromise?

Reviewer: I have added the suggestion in another review. Will resolve this again.

# Update CRDs to nightly version (on main)
# Repeat this for every operator used by the demo

> kubectl replace -f https://raw.githubusercontent.com/stackabletech/commons-operator/main/deploy/helm/commons-operator/crds/crds.yaml

Reviewer: Maybe add in a placeholder to prevent auto-brain from only doing one thing.

```suggestion
kubectl replace -f https://raw.githubusercontent.com/stackabletech/commons-operator/main/deploy/helm/commons-operator/crds/crds.yaml
kubectl replace -f https://raw.githubusercontent.com/stackabletech/...-operator/main/deploy/helm/...-operator/crds/crds.yaml
```

> # Install dev version of operators

Reviewer: I don't mind if we say nightly or dev, but we should be consistent.

```suggestion
# Install nightly version of operators
```

> # Install dev version of operators
> stackablectl operator install commons ...

Reviewer: ```suggestion
# Install dev version of operators (use the list from the earlier step before deleting the operators)
stackablectl operator install commons ...
```

Author: Done in 3a9ef57

> # Optionally update the product versions in the CRDs, e.g.:
> kubectl edit hbaseclusters/hbase

Reviewer: ```suggestion
# Optionally update the product versions in the CRDs, e.g.:
kubectl patch hbaseclusters/hbase --type='json' -p='[{"op": "replace", "path": "/spec/image/productVersion", "value":"x.x.x"}]' # changed

```

-->

<!--
    DO NOT REMOVE THIS COMMENT. It is intended for people who might copy/paste from the previous release issue.
    This was created by an issue template: https://github.com/stackabletech/issues/issues/new/choose.
-->

## Pre-Release Demo Testing on Nightly

Part of <https://github.com/stackabletech/issues/issues/TRACKING_ISSUE>

For each demo, run the following commands:

> [!NOTE]
> Record any issues or anomalies during the process in a comment on this issue.
> Eg:
> ```
> :green_circle: **airflow-scheduled-job**
>
> The CRD had been updated and I needed to change the following in the manifest:
> ...
> ```

```shell
# Install demo (stable)
stackablectl demo install <DEMO_NAME>

# Get a list of installed operators
stackablectl operator installed

# Sometimes it is necessary to upgrade Helm charts
# First, add the Helm repositories
helm repo add minio https://charts.min.io/
helm repo add bitnami https://charts.bitnami.com/bitnami

# Next, upgrade the charts, i.e.:
helm upgrade minio minio/minio --version x.x.x
helm upgrade postgresql-hive bitnami/postgresql --version x.x.x
helm upgrade postgresql-hive-iceberg bitnami/postgresql --version x.x.x

> helm upgrade postgresql-superset bitnami/postgresql --version x.x.x

Author: Yes, but how do we want to solve this? Mention that these differ based on the demo? Or do you have something else in mind?

> helm upgrade postgresql-superset bitnami/postgresql --version x.x.x

Reviewer: I don't think we have to completely solve it (as in automatically find things), but adding some hints would help. Eg:

```sh
# Sometimes it is necessary to upgrade Helm charts. Look for other Helm Charts
# which might need updating.

# First, see which charts are installed. You can ignore the stackable-operator charts, or anything that might have been installed outside of this demo.
helm list

# Next, add the applicable Helm Chart repositories. For example:
helm repo add minio https://charts.min.io/
helm repo add bitnami https://charts.bitnami.com/bitnami

# Finally, upgrade the Charts. For example:
helm upgrade minio minio/minio --version x.x.x
helm upgrade postgresql-hive bitnami/postgresql --version x.x.x
```

> helm upgrade postgresql-superset bitnami/postgresql --version x.x.x

Author: Changed in 70794e3
